CppStudy/CppStudy.cpp

```diff
@@ -4,59 +4,111 @@
-    std::cout << std::format("Number: {}", employee.employeeNumber) << std::endl;
-    std::cout << std::format("Salary: {}", employee.salary) << std::endl;
-
-    std::cout << "Exercise 1-2 -----------------------------\n";
-    std::cout << format("Employee title: {}", getTitleName(employee.title)) << std::endl << std::endl;
-
-    std::cout << "Exercise 1-3 -----------------------------\n";
-    std::array<HR::Employee, 3> employees{
-        HR::Employee{'A', 'X', 1, 60000, HR::Title::ENGINEER},
-        HR::Employee{'B', 'Y', 1, 80000, HR::Title::SENIOR_ENGINEER},
-        HR::Employee{'A', 'Z', 1, 100000, HR::Title::MANAGER}
-    };
-
-    for (const auto& employee : employees) {
-        printEmployee(employee);
-    }
-
-    std::cout << "Exercise 1-4 -----------------------------\n";
-    std::vector<HR::Employee> vecEmployees;
-    for (const auto& employee : employees) {
-        vecEmployees.push_back(employee);
-    }
-
-    std::cout << "Exercise 1-5, 1-6 ------------------------\n";
-    AirlineTicket airlineTicket{ "Fjodor Sumkin", 137, true };
-    airlineTicket.setFrequentFlyerNumber(555);
-
-    std::cout << std::format("Ticket for {}, miles: {}, has rewards: {}",
-        airlineTicket.getPassengerName(), airlineTicket.getNumberOfMiles(),
-        airlineTicket.hasEliteSuperRewardStatus()) << std::endl;
-
-    if (airlineTicket.getFrequentFlyerNumber()) {
-        std::cout << std::format("Frequent Flyer Number: {}", airlineTicket.getFrequentFlyerNumber().value()) << std::endl;
-    }
+	excersise7_2();
+	excersise7_3();
+	excersise7_4();
 
     return 0;
 }
+
+void excersise7_1()
+{
+	const size_t numberOfElements{ 10 };
+	int* values{ new int[numberOfElements] };
+
+	/**
+	* 1. There is no need to create an array on the heap in this case since number of elements is a constant value
+	* 2. Could be used std::array instead of using raw pointers
+	* 3. Could be wrapped in unique ptr if the heap allocation is still considered
+	*/
+
+	// Set values to their index value.
+	for (int index{ 0 }; index < numberOfElements; ++index) {
+		values[index] = index;
+	}
+
+	/**
+	* 1. This can be done using special tools
+	*/
+	
+	// Set last value to 99.
+	values[10] = 99;
+
+	/**
+	* 1. Last element index is 9, not 10. This will set the memory out of array bounds
+	* 2. Better to use "numberOfElements - 1" instead of hardcoded index value
+	* 3. Better to use size_t for the loop iterator type
+	*/
+	// Print all values.
+	for (int index{ 0 }; index <= numberOfElements; ++index) {
+		std::cout << values[index] << " ";
+	}
+
+	/**
+	* 1. This will address array out of bounds again when index is equal to numberOfElements
+	* 2. Raw pointer is not deleted in the end of use, memory leak!
+	*/
+}
+
+void excersise7_2()
+{
+	std::array<int, 10> values {};
+
+	std::iota(values.begin(), values.end(), 0);
+
+	// Set last value to 99.
+	values[values.size() - 1] = 99;
+
+	// Print all values.
+	for (size_t i{ 0 }; i < values.size(); ++i) {
```

Comment:
it's preferable to not iterate using conter.
it's better to use:
- std::ranges if you still need an index
- for (auto item : values)
- for (auto iterator = values.begin(); iterator != values.end(); ++iterator)
- 
maybe there are even more options....